py-polars/polars/utils/convert.py

```diff
@@ -184,26 +180,26 @@ def _to_python_datetime(
 ) -> datetime:
     """Convert polars int64 timestamp to Python datetime."""
     if not time_zone:
-        if time_unit == "us":
+        if time_unit == "ns":
+            return EPOCH + timedelta(microseconds=value // 1_000)
+        elif time_unit == "us":
```

Comment:
The order should be `us` -> `ns` -> `ms` in these if-statement chains, due to `us` being the most common.

Reply:
Should the `if-else` order you suggested ( `us `-> `ns` -> `ms`) be implemented in the other methods; `_datetime_to_pl_timestamp`, `_timedelta_to_pl_timedelta`?